browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -356,5 +367,100 @@ void BraveTabContainer::PaintChildren(const views::PaintInfo& paint_info) {
   }
 }
 
+BrowserRootView::DropIndex BraveTabContainer::GetDropIndex(
+    const ui::DropTargetEvent& event) {
+  if (!base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs) ||
+      !tabs::utils::ShouldShowVerticalTabs(
+          tab_slot_controller_->GetBrowser())) {
+    return TabContainerImpl::GetDropIndex(event);
+  }
+
+  // Force animations to stop, otherwise it makes the index calculation tricky.
+  CompleteAnimationAndLayout();
+
+  const int x = GetMirroredXInView(event.x());
+  const int y = event.y();
+
+  std::vector<TabSlotView*> views = layout_helper_->GetTabSlotViews();
+
+  // Loop until we find a tab or group header that intersects |event|'s
+  // location.
+  for (TabSlotView* view : views) {
+    const int max_y = view->y() + view->height();
+    if (y >= max_y) {
+      continue;
+    }
+
+    if (view->GetTabSlotViewType() == TabSlotView::ViewType::kTab) {
+      Tab* const tab = static_cast<Tab*>(view);
+
+      // Closing tabs should be skipped.
+      if (tab->closing()) {
+        continue;
+      }
+
+      if (tab->data().pinned && !IsPointInTab(tab, gfx::Point(x, y))) {
+        continue;
+      }
+
+      // When a tab is dropped between pinned tabs, it will add a new tab
+      // among unpinned tabs. As a result, there's no need to check the hot
+      // height for pinned tabs.
+      const int model_index = GetModelIndexOf(tab).value();
+      if (!tab->data().pinned) {
+        const bool first_in_group =
+            tab->group().has_value() &&
+            model_index ==
+                controller_->GetFirstTabInGroup(tab->group().value());
+
+        // When hovering over the top or bottom quarter of a tab, the drop
+        // indicator will point between tabs.
+        const int hot_height = tab->height() / 4;
+
+        if (y >= (max_y - hot_height)) {
+          return {model_index + 1, true /* drop_before */,
+                  false /* drop_in_group */};
+        }
+
+        if (y < tab->y() + hot_height) {
+          return {model_index, true /* drop_before */, first_in_group};
+        }
+      }
+
+      return {model_index, false /* drop_before */, false /* drop_in_group */};
+
+    } else {
+      TabGroupHeader* const group_header = static_cast<TabGroupHeader*>(view);
+      const int first_tab_index =
+          controller_->GetFirstTabInGroup(group_header->group().value())
+              .value();
+
+      return {first_tab_index, true /* drop_before */,
+              y >= max_y - group_header->height() / 2 /* drop_in_group */};
+    }
+  }
+
+  // The drop isn't over a tab, add it to the end.
+  return {GetTabCount(), true, false};
+}
+
+// TODO: Add functionality for 'HandleDragUpdate' and 'HandleDragExited'
+// during the implementation of a drop arrow when dragging and dropping text
+// or a link.
+void BraveTabContainer::HandleDragUpdate(
```

Comment:
Q. I'm curious what blocks you from going further with this. Is there something you need?

Reply:
The existing drop arrow operates on the x-axis. We intend to adapt it for the y-axis in the case of vertical tabs and implement the appropriate logic for drop arrows on expanded pinned tabs. This task may take some time, and to avoid complicating this pull request, I'll address it after a thorough review and successful merge of this current PR.

Reply:
If you don't mind, it'd be better to do that in this PR together as without the arrow indicator, users can't find out what will happen when dropping the url - load url on the existing tab or create new tab with the URL. I believe we don't want to riddle users 😄

Reply:
@sangwoo108 I require the drop arrow(right, left) resource. Could you share it with me, or should I transform the Chromium icon and use it for our purpose?

Reply:
Yeah, please show the arrow! 👍🏼